File: app/src/main/java/com/mapbox/mapboxsdk/plugins/testapp/activity/places/PickerLauncherActivity.java
package com.mapbox.mapboxsdk.plugins.testapp.activity.places;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AppCompatActivity;
import android.widget.Toast;

import com.mapbox.api.geocoding.v5.models.CarmenFeature;
import com.mapbox.mapboxsdk.Mapbox;
import com.mapbox.mapboxsdk.camera.CameraPosition;
import com.mapbox.mapboxsdk.geometry.LatLng;
import com.mapbox.mapboxsdk.plugins.testapp.R;
import com.mapbox.plugins.places.picker.PlacePicker;
import com.mapbox.plugins.places.picker.model.PlacePickerOptions;

import butterknife.ButterKnife;
import butterknife.OnClick;

public class PickerLauncherActivity extends AppCompatActivity {

  private static final int REQUEST_CODE = 5678;

  @Override
  public void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_picker_launcher);
    ButterKnife.bind(this);
  }

  @OnClick(R.id.launch_location_picker)
  public void onClick(FloatingActionButton view) {
    startActivityForResult(
      new PlacePicker.IntentBuilder()
        .accessToken(Mapbox.getAccessToken())
        .placeOptions(PlacePickerOptions.builder()
          .statingCameraPosition(new CameraPosition.Builder()
            .target(new LatLng(40.7544, -73.9862)).zoom(16).build())
          .build())
        .build(this), REQUEST_CODE);
  }

  @Override
  protected void onActivityResult(int requestCode, int resultCode, Intent data) {
    super.onActivityResult(requestCode, resultCode, data);
    if (requestCode == REQUEST_CODE && resultCode == RESULT_OK) {
      CarmenFeature carmenFeature = PlacePicker.getPlace(data);
      String toastMsg = String.format("Place: %s", carmenFeature.placeName());
      Toast.makeText(this, toastMsg, Toast.LENGTH_LONG).show();
    }
  }
}

File: app/src/main/java/com/mapbox/mapboxsdk/plugins/testapp/activity/places/PickerLauncherActivity.kt
package com.mapbox.mapboxsdk.plugins.testapp.activity.places

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.mapbox.mapboxsdk.Mapbox
import com.mapbox.mapboxsdk.camera.CameraPosition
import com.mapbox.mapboxsdk.geometry.LatLng
import com.mapbox.mapboxsdk.plugins.places.picker.PlacePicker
import com.mapbox.mapboxsdk.plugins.places.picker.model.PlacePickerOptions
import com.mapbox.mapboxsdk.plugins.testapp.R
import kotlinx.android.synthetic.main.activity_picker_launcher.*

class PickerLauncherActivity : AppCompatActivity() {

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_picker_launcher)
        reverseGeocodingSwitch.text = getString(R.string.reverse_geocoding_disabled)
        reverseGeocodingSwitch.setOnCheckedChangeListener { compoundButton, checked ->
            reverseGeocodingSwitch.text = if (checked)
                getString(R.string.reverse_geocoding_enabled)
            else getString(R.string.reverse_geocoding_disabled)
        }

        userLocationSwitch.text = getString(R.string.user_location_button_disabled)
        userLocationSwitch.setOnCheckedChangeListener { compoundButton, checked ->
            userLocationSwitch.text = if (checked)
                getString(R.string.user_location_button_enabled)
            else getString(R.string.user_location_button_disabled)
        }

        fabLocationPicker.setOnClickListener { _ ->
            Mapbox.getAccessToken()?.let {
                startActivityForResult(
                        PlacePicker.IntentBuilder()
                                .accessToken(it)
                                .placeOptions(PlacePickerOptions.builder()
                                        .includeReverseGeocode(reverseGeocodingSwitch.isChecked)
                                        .includeDeviceLocationButton(userLocationSwitch.isChecked)
                                        .statingCameraPosition(CameraPosition.Builder()
                                                .target(LatLng(40.7544, -73.9862))
                                                .zoom(16.0)
                                                .build())
                                        .build())
                                .build(this), REQUEST_CODE)
            }
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == REQUEST_CODE && resultCode == Activity.RESULT_OK) {
            if (reverseGeocodingSwitch.isChecked) {
                val carmenFeature = PlacePicker.getPlace(data)
                Toast.makeText(this, carmenFeature?.placeName(), Toast.LENGTH_LONG).show()
            } else {
                val cameraPosition = PlacePicker.getLastCameraPosition(data)
                Toast.makeText(this, cameraPosition.target.toString(), Toast.LENGTH_LONG).show()
            }
        }
    }

    companion object {
        private val REQUEST_CODE = 5678
    }
}
